internal/azure/cache.go

```diff
@@ -0,0 +1,80 @@
+// Package azure provides Azure SDK integration for AKS MCP server.
+package azure
+
+import (
+	"sync"
+	"time"
+)
+
+// Cache is a simple in-memory cache for Azure resources.
+type AzureCache struct {
+	data           map[string]cacheItem
+	mu             sync.RWMutex
+	defaultTimeout time.Duration
+}
+
+// cacheItem represents a cached resource with expiration time.
+type cacheItem struct {
+	value      interface{}
+	expiration time.Time
```

Comment:
seems all is defaultTimeout? is it worth a new field?

Reply:
it is used for **each item**  to know exactly when it expires, if we don't have this new field, we need to have another filed like 'createdTime'. so having this new field is added on purpose.